homeassistant/components/zha/repairs.py

```diff
@@ -0,0 +1,126 @@
+"""ZHA repairs for common environmental and device problems."""
+from __future__ import annotations
+
+import enum
+import logging
+
+from universal_silabs_flasher.const import ApplicationType
+from universal_silabs_flasher.flasher import Flasher
+
+from homeassistant.components.homeassistant_sky_connect import (
+    hardware as skyconnect_hardware,
+)
+from homeassistant.components.homeassistant_yellow import (
+    RADIO_DEVICE as YELLOW_RADIO_DEVICE,
+    hardware as yellow_hardware,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import issue_registry as ir
+
+from .core.const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AlreadyRunningEZSP(Exception):
+    """The device is already running EZSP firmware."""
+
+
+class HardwareType(enum.StrEnum):
+    """Detected Zigbee hardware type."""
+
+    SKYCONNECT = "skyconnect"
+    YELLOW = "yellow"
+    OTHER = "other"
+
+
+DISABLE_MULTIPAN_URL = {
+    HardwareType.YELLOW: (
+        "https://yellow.home-assistant.io/guides/disable-multiprotocol/#flash-the-silicon-labs-radio-firmware"
+    ),
+    HardwareType.SKYCONNECT: (
+        "https://skyconnect.home-assistant.io/procedures/disable-multiprotocol/#step-flash-the-silicon-labs-radio-firmware"
+    ),
+    HardwareType.OTHER: None,
+}
+
+ISSUE_WRONG_SILABS_FIRMWARE_INSTALLED = "wrong_silabs_firmware_installed"
+
+
+def detect_radio_hardware(hass: HomeAssistant, device: str) -> HardwareType:
```

Comment:
We already have something similar to this in the `otbr` and `zha` config flows, including the explicit imports of the hardware platforms from `homeassistant_yellow` and `homeassistant_sky_connect`.

Could we refactor the code so there's either either one implementation, instead of three, which checks if a device path points to either a `homeassistant_yellow` or a `homeassistant_sky_connect`, or it's made trivial to check if a device belongs to a Home Assistant Yellow's Zigbee radio or to a Home Assistant SkyConnect?

Maybe:
- Update `HardwareInfo` to include the domain of the integration which has the hardware platform
- Add a new member `radio` to `HardwareInfo` which has the device path and type of a radio
- Make the  `hardware` integration export a function which returns `list[HardwareInfo]`, for example by moving the implementation of the "hardware/info" WS command out of the WS handler to a public utility function.

This can be fixed in a follow-up PR.